### PR TITLE
Unify styles by refactoring Sass into Emotion.js / CSS-in-JS

### PR DESCRIPTION
**Pull request by jonkafton**
# What are the relevant tickets?

Adresses #239

# Description (What does it do?)

This is an initial PR to serve as a reference for ongoing work to unify how styles are applied by replacing CSS/Sass declarations (.scss) files with CSS-in-JS, implemented with [Emotion](https://emotion.sh/).

Included are replacement functionality for theming and media query breakpoints and style refactor for the header and home page.

* Adds Emotion.js to the mit-open workspace

* Replicates the global theme scss with object styles

* Creates styled components for the header and its child components

* Creates media query breakpoint functionality to replace scss

* Applies styled components to the home page

* Creates media query breakpoints functionality to replace scss

* Applies styled components to the home page

# How can this be tested?

Changes are non functional. Unit tests must pass, though otherwise tested by eye (perhaps there's a case to introduce screenshot diff testing).

### CODE REVIEW

File: frontends/ol-util/src/styles/breakpoints.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+export const mediaQueries = {
+  down: (breakpoint: keyof typeof breakpoints) => {
+    return `@media (max-width: ${breakpoints[breakpoint]}px)`
+  },
+  up: (breakpoint: keyof typeof breakpoints) => {
+    return `@media (min-width: ${breakpoints[breakpoint]}px)`
+  },
+}
+console.log("mediaQueries", mediaQueries)
```
Comment (ChristopherChudzicki) [inline frontends/ol-util/src/styles/breakpoints.ts:27]:
```suggestion

```

**Comment by jonkafton** [comment]
Thanks @ChristopherChudzicki. I agree we we should ideally define our theme in one place.

The idea is that we currently a have top level application theme defined in Sass, so theme.ts is intended to replace that.

We can extend the MUI theme options to set these app level themes, though they are not compatible with Material UI's theme specification so would be shoehorned. I'll code this up in any case so we can see it as an option - commit upcoming.

It may make sense to do so if we were fully invested in MUI, though if we're only considering it or are progressively adding it in, it will make a cleaner distinction for now to have the two theme providers.

The key question is whether the theme is unique to this MIT Open app, or whether we want it to be reused in other projects. I see the theme as top level to the app, in which case it should be passed to the ol-design component, rather than defined there. I'm not sure about using ol-design to provide top level styles as this breaks the idea of moving the styles close to their components - page level styles would then live in the mit-open workspace. If theming is intended for reuse in other projects, we could give it a name (MITOpenTheme?) and provide it for import from ol-design to be passed into the MUI theme options, though I think ol-design should be a reusable component catalog (rename to ol-components?); baking in specific theme values there discourages reuse.

I'm thinking here with the overall structural rework in mind, which I've written up here, [MIT Open - Front End Component Structure](https://docs.google.com/document/d/1AVeolsY1ezkoNIZpyRLIjYI5XB4JAhFRj2TuXh3h8fQ), so would like to get your thoughts on that as this is closely linked.

**Comment by jonkafton** [comment]
This commit https://github.com/mitodl/mit-open/pull/245/commits/37ab89b augments the MUI theme with the values in mit-open/src/scss/theme.scss so they are imported via ol-design.

These are not compatible with the [MUI theme schema](https://mui.com/material-ui/customization/theming/#theme-configuration-variables), so are added under a `custom` property. We may want to migrate the page level theme to match MUI, though it's quite specific (assumes the [Material Design color system](https://m3.material.io/styles/color/system/overview)) and will depend on our design requirements as they emerge. In any case, we will want to do this progressively alongside any restructuring.

I've updated the linked issue https://github.com/mitodl/mit-open/issues/239 to capture thoughts.

**Review by ChristopherChudzicki — CHANGES_REQUESTED** [review]
This looks good 👍 

One request to remove `xsDown`, etc.

And a suggestion about typecsript that I think is worthwhile.

File: frontends/mit-open/src/components/Header.tsx [hunk 2]
```diff
@@ -1,16 +1,46 @@
...
 
-const Header: React.FC = () => {
+interface BarProps {
+  theme?: Theme
+  position: string
+}
+
+const Bar = styled(AppBar)<BarProps>`
+  background-color: ${({ theme }) => theme.custom.colorBackgroundLight};
+  color: ${({ theme }) => theme.custom.fontColorDefault};
```
[the remark this answers is not in the document]
Comment (ChristopherChudzicki) [reply frontends/mit-open/src/components/Header.tsx:9]: > I did find I needed to give the compiler a triple slash reference to find it in mit-open/src/components/Header.tsx, though not pages/Home/Home.tsx (?!) 

We won't need it if the triple slash in `frontends/ol-design/src/index.ts` is at the top of the file. Triple slash directives are only valid at the top of the file. (Which I missed in the [TS docs](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html) when I first tried this!)

> I've added a rule to the ESLint config to allow triple slash directives for types only, with prefer-import
👍 I think that's fine. `import type` is better where it works, but this I don't think people will use the triple slash unnecessarily. UsuallyI forget it exists, except when I run into the exact case where it's still necessary.

**Review by ChristopherChudzicki — CHANGES_REQUESTED** [review]
Looks good except can we move the triple slash in `frontends/ol-design/src/types` to top of file (so it's valid) and remove the one in Header.tsx.

**Pull request merged**